src/lib/featureFlags/featureFlagProvider.ts

```diff
@@ -0,0 +1,12 @@
+export type FeatureFlag = string
+export type FeatureFlags = Array<string>
+
+export interface FeatureFlagConfig extends Iterable<string> {}
+
+export type FeatureFlagContext = Record<string, any>
+
+export interface FeatureFlagProvider {
```

Comment:
As this file is just a collection of types and not the actual provider, that should be noted in the filename -- `featureFlagTypes.ts`.